lib/src/identity_codec.dart

```diff
@@ -0,0 +1,29 @@
+import 'dart:convert';
+
+class _IdentityConverter<T> extends Converter<T, T> {
+  @override
+  T convert(T input) => input;
+}
+
+/// A [Codec] that converts from T to T doing absolutely nothing.
+///
+/// This is the identity codec, the one that passes input to output when
+/// decoding and encoding. This codec is mostly useful when implementing generic
+/// code that can be fused with a [Codec] and you need an identity codec for the
+/// base case.
+///
+/// Note, that when fused with another [Codec] the identity codec disppears.
+class IdentityCodec<T> extends Codec<T, T> {
+  const IdentityCodec();
+
+  @override
+  Converter<T, T> get decoder => _IdentityConverter();
+  @override
+  Converter<T, T> get encoder => _IdentityConverter();
+
+  /// Fuse with an other codec.
+  ///
+  /// This will return the [other] codec, as the identity codec is trivial.
+  @override
+  Codec<T, R> fuse<R>(Codec<T, R> other) => other;
```

Comment:
This is not handling the case of `other.fuse(identityCodec)`.

On the other hand, I don't want the generic `fuse` to know about `IdentifyCodec` (it could mean that the class is harder to tree-shake).

Reply:
We could document this behavior, but that seems overkill.. And cost of doing it wrong is likely small anyways.